modules/videopress/class.videopress-gutenberg.php

```diff
@@ -164,6 +165,28 @@ public function render_video_block_with_videopress( $attributes, $content ) {
 			1
 		);
 	}
+
+	/**
+	 * Replaces the video uploaded in the block editor.
+	 *
+	 * Enqueues a script that registers an API fetch middleware replacing the video uploads in Gutenberg so they are
+	 * uploaded against the WP.com API media endpoint and thus transcoded by VideoPress.
+	 */
+	public function override_video_upload() {
+		if (
```

Comment:
Can we also make this conditional on it being a REST API request, and bail early on the failure instead? I think we can use `REST_REQUEST` here. That way, the script isn't unnecessarily loaded in other contexts.

Reply:
Uhm, we need to load this script on the WP Admin editor, not on an API context. This is a JS file that will intercept any `wp.apiFetch` call to `/wp/v2/media` and swap it out to `public-api.wordpress.com/rest/v1.1/media/new`.